Ginger/GingerCoreCommon/VariablesLib/VariableSelectionList.cs

```diff
@@ -121,14 +124,53 @@ public override void ResetValue()
         {
             if (OptionalValuesList.Count > 0)
             {
+                OptionalValuesList.CurrentItem = OptionalValuesList.ElementAt<OptionalValue>(0);
                 Value = OptionalValuesList[0].Value;
             }
 
         }
 
-        public override void GenerateAutoValue()
+        public override void GenerateAutoValue(ref string errorMsg)
         {
-            //NA
+            if (OptionalValuesList.Count == 0)
+            {
+                Value = string.Empty;
+                errorMsg = "Generate Auto Value is not possible because Selection List is empty";
+                return;
+            }
+
+            int index = OptionalValuesList.IndexOf((OptionalValue)OptionalValuesList.CurrentItem);
+
+            //Check if the OptionalValue is last
+            if (index == OptionalValuesList.Count - 1)
+            {
+                //If loop chechbox is disabled so return a proper message.
+                if (!IsLoopEnabled)
+                {
+                    errorMsg = "Generate Auto Value is not possible because current value is last and looping is not allowed";
+                }
+                else
+                {
+                    OptionalValuesList.CurrentItem = OptionalValuesList.ElementAt<OptionalValue>(0);
```

Comment:
'OptionaValue'? also i'm not convinced we need to use CurrentItem and MoveNext, let's discuss on it